src/components/Notification.tsx

```diff
@@ -0,0 +1,36 @@
+import { FC } from 'react';
+import classNames from 'classnames';
+import { NotificationType } from '../types/NotificationType';
+import { useGlobalContext } from '../hooks/useGlobalContext';
+
+type Props = {
+  type: NotificationType;
+} & (
+  {
+    type: NotificationType.PostsWarning;
+    warningText: string
+  } | {
+    type: Exclude<NotificationType, NotificationType.PostsWarning>;
+    warningText?: string;
+  }
+);
+
+export const Notification: FC<Props> = ({ type, warningText }) => {
+  const { error } = useGlobalContext();
```

Comment:
It's a good practice to handle all possible states by providing a loading and error state. In this case, when error state occurs, it's handled, but there is no loading state considered.